src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -163,14 +170,24 @@ public void updateFilteredPersonList(Predicate<Person> predicate) {
         requireNonNull(predicate);
         filteredPersons.setPredicate(predicate);
     }
-
+    /**
+     * Updates an unmodifiable view of the list of {@code Event} backed by the internal list of
+     * {@code versionedAddressBook}
+     */
     @Override
     public void updateFilteredEventList(Predicate<Event> predicate) {
         requireNonNull(predicate);
         filteredEvents.setPredicate(predicate);
     }
 
-
+    /**
+     * Updates the reference between Event and the persons it is tagged with.
+     * This method is used after editPerson, deletePerson and tagEvent commands to provide timely GUI update.
+     */
+    @Override
+    public void updateEventPersonReference() {
+        this.updateFilteredEventList(x -> x.getUids().setPersonNames(this));
```

Comment:
Yes I am trying to update the person names in every event and show all events by the boolean method `setPersonNames()`, that's why it always returns `true`, so that all events with updated person names will be displayed. I agree with you that it's clearer to split it and I'll try doing it in 2 steps.

Reply:
I just tried splitting this method into 2 steps, the backend works fine, but there's a problem with the GUI update. The person names under each event doesn't update immediately after I input a tagEvent command, it will only update after I input another command or click on the edited event. Therefore I think it's better to leave this method as it is now?

Reply:
Yeah I guess so, then perhaps include a comment to explain what it does